### PR TITLE
Docs: uasyncio.rst Detail exceptions in cancellation and timeouts

### PR DESCRIPTION
**Pull request by peterhinch**
Replaces https://github.com/micropython/micropython/pull/5883. Clarify the exceptions which are thrown in task cancellation and timeout.

### CODE REVIEW

File: docs/library/uasyncio.rst [hunk 1]
```diff
@@ -68,7 +68,9 @@ Additional functions
     created from it.
```
Comment (tve) [inline docs/library/uasyncio.rst:68]: also fix /that/than/ in "longer that *timeout* seconds"?

File: docs/library/uasyncio.rst [hunk 2]
```diff
@@ -68,7 +68,9 @@ Additional functions
...
 
     If a timeout occurs, it cancels the task and raises ``asyncio.TimeoutError``:
-    this should be trapped by the caller.
+    this should be trapped by the caller.  The task receives
+    ``asyncio.CancelledError`` which may be ignored.  Cleanup code may be run by
```
Comment (tve) [inline docs/library/uasyncio.rst:72]: Not 100% obvious where the cleanup code is meant to be... Maybe rephrase slightly to "... which may be ignored or trapped using `try...except`/`try...finally` to run cleanup code".

**Comment by peterhinch** [comment]
Fair point - now done.

**Comment by dpgeorge** [comment]
Thanks for this, and sorry it got completely lost.  Merged in d94ac4333ff87e8a1b28ae3b10b1ef0c6de9cfb0
